src/main/scala/dpla/ingestion3/confs/OaiHarvesterConf.scala

```diff
@@ -22,43 +20,53 @@ import dpla.ingestion3.utils.Utils.validateUrl
   * sparkMaster (Optional): If omitted the defaults to local[*]
   */
 
-class OaiHarvesterConf(arguments: Seq[String]) extends ScallopConf(arguments) {
-  // Required parameters
-  val outputDir: ScallopOption[String] = opt[String]("outputDir",
-    required = true,
-    noshort = true,
-    validate = _.nonEmpty)
-  val endpoint: ScallopOption[String] = opt[String]("endpoint",
-    required = true,
-    noshort = true)
-  val verb: ScallopOption[String] = opt[String]("verb",
-    required = true,
-    noshort = true)
-  val provider: ScallopOption[String] = opt[String]("provider",
-    required = true,
-    noshort = true)
-  // Optional parameters
-  val prefix: ScallopOption[String] = opt[String]("prefix",
-    required = false,
-    noshort = true,
-    validate = _.nonEmpty)
-  val harvestAllSets: ScallopOption[String] = opt[String]("harvestAllSets",
-    required = false,
-    noshort = true,
-    default = Some("false"))
-  val setlist: ScallopOption[String] = opt[String]("setlist",
-    required = false,
-    noshort = true,
-    validate = _.nonEmpty)
-  val blacklist: ScallopOption[String] = opt[String]("blacklist",
-    required = false,
-    noshort = true,
-    validate = _.nonEmpty)
-  // If a master URL is not provider then default to local[*]
-  val sparkMaster: ScallopOption[String] = opt[String]("sparkMaster",
-    required = false,
-    noshort = true,
-    default = Some("local[*]"))
+class OaiHarvesterConf(arguments: Seq[String]) {
+  def load(): OaiConfig = {
+    // Loads config file for more information about loading hierarchy please read
+    // https://github.com/typesafehub/config#standard-behavior
+    ConfigFactory.invalidateCaches()
+    val oaiConf = ConfigFactory.load()
+
+    def getProp(prop: String, default: Option[String] = None): Option[String] = {
+      oaiConf.hasPath(prop) match {
```

Comment:
That part is built in, right? What I'm suggesting is the possibility of ad hoc overriding a property with an environment variable named the same thing, i.e., `export oaiParams.endpoint=http://example.com && sbt run`

We had talked about that as an easy way to achieve what @markbreedlove  was asking for without having to write a lot of argument parsing code alongside the config code.